daemon/container.go

```diff
@@ -1421,7 +1421,11 @@ func (container *Container) startLogging() error {
 		}
 		l = dl
 	case "syslog":
-		dl, err := syslog.New(container.ID[:12])
+		tag := container.ID[:12]
+		if len(container.Name) > 1 {
```

Comment:
I've seen odd situation where I end up with a container with no name.  I figured it was best to be defensive in this situation.  If you think this is overkill I can just change this code to always do `tag := container.Name[1:]`